crates/turbopack/src/module_options/module_rule.rs

```diff
@@ -23,8 +24,8 @@ impl ModuleRule {
         self.effects.iter()
     }
 
-    pub fn matches(&self, path: &FileSystemPath, reference_type: &ReferenceType) -> bool {
```

Comment:
Nit: I think we should still pass in the `&FileSystemPath`, since most rules just need that. We'd be able to skip the async delay for most things (I'm assuming a an async function that doesn't `.await` anything is measurably faster than one that does, even if that `.await` is immediately ready).